tests/test_smtlibv2.py

```diff
@@ -262,34 +263,57 @@ def testBasicArrayConcatSlice(self):
             results.append(c)
         self.assertTrue(len(results) == 5)
 
+    def testBasicArraySlice(self):
+        hw = bytearray(b'Hello world!')
+        cs =  ConstraintSet()
+        #make array of 32->8 bits
+        array = cs.new_array(32, index_max=12)
+        array = array.write(0, hw)
+        array_slice = array[0:2]
+        self.assertTrue(self.solver.must_be_true(cs, array == hw))
+        self.assertTrue(self.solver.must_be_true(cs, array_slice[0] == array[0]))
+        self.assertTrue(self.solver.must_be_true(cs, array_slice[0:2][1] == array[1]))
+        array_slice[0] = ord('A')
+        self.assertTrue(self.solver.must_be_true(cs, array_slice[0] == ord('A')))
+        self.assertTrue(self.solver.must_be_true(cs, array_slice[0:2][1] == array[1]))
```

Comment:
Wondering, what happens if I do:
```
array_slice[0:10000]
```
when the slice has only 2 elements? Will it return me a slice of those two, as with Python lists?

We might want add a testcase for it.

Reply:
Some tests added.